### PR TITLE
multiple owners for tokens

### PR DESCRIPTION
**Pull request by rbns**

### CODE REVIEW

File: app/controllers/api_tokens_controller.rb [hunk 1]
```diff
@@ -33,8 +33,9 @@ def edit
 
   def update
     @api_token = ApiToken.find(params[:id])
+    puts params
```
Comment (fkr) [inline app/controllers/api_tokens_controller.rb]: Debug output that snuck in?

**Comment by fkr** [comment]
Aside from the debug output it looks good to me. @dra any objections?

File: app/controllers/api_tokens_controller.rb (hunk 1 above)
Comment (rbns) [reply app/controllers/api_tokens_controller.rb]: indeed, i removed it now.

**Comment by dra** [comment]
The code looks OK to me.

Not sure if there should be a migration connecting all existing tokens with all existing owners, to remain the status quo: All tokens have access to all areas.
Otherwise all existing tokens are useless until manually connected to owners.

**Comment by fkr** [comment]
@dra: very good point!

Imho the migration should clear away the existing tokens and the updating notes should contain the note that it is necessary to add the tokens again with explicit owner assignment.

**Comment by rbns** [comment]
my two cents:
keeping the existing tokens saves the work for readding them, though that shouldn't be too much work. adapters would receive empty data when using these tokens, which may be no obvious failure mode.

removing them completely results in a more obvious failure. if the tokens aren't recreated, the adapters can't authenticate anymore and get 401 unauthorized.

an "active" flag for tokens would be great now which could be set to false for existing tokens. that way no existing tokens have to be removed and adapters would get a clean failure message.

opinion?

**Comment by rbns** [comment]
we won't do this for now

**Comment by rbns** [comment]
a bit more reasoning why this is closed:

- it would be complicated to get a good mapping from api clients to owners. it would either require the api client to know about the owners, or to have some functionality in the idb to map items submitted via api to owners, like fqdn-regexps or similar. either of these ways is more complicated than useful and likely to break.

- the functionality can be achieved using the tokens like they are now, one owner per token. the only requirement for this is that api clients can be configured to have multiple api tokens to use. for example idbiaas has a idb configuration per zone: https://github.com/idb-project/idbiaas#example-configuration
with this, either a "master" token and owner is required to fetch the configuration for idbiaas from the idb, or the configuration is stored locally.

**Pull request closed without merging**